core/src/main/java/tech/pegasys/eth2signer/core/multikey/ImmutableArtifactSignerProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.multikey;
+
+import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
+import tech.pegasys.eth2signer.core.signing.ArtifactSignerProvider;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ImmutableArtifactSignerProvider implements ArtifactSignerProvider {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Map<String, ArtifactSigner> signers;
+
+  private ImmutableArtifactSignerProvider(final Map<String, ArtifactSigner> signers) {
+    this.signers = signers;
+  }
+
+  public static ImmutableArtifactSignerProvider wrap(final Collection<ArtifactSigner> signers) {
```

Comment:
create instead of wrap? This is really the constructor it's not really wrapping as it's converting from a collection to a map.

Reply:
done